events/bus_test.go

```diff
@@ -155,67 +120,8 @@ func TestBus_Subscribe(t *testing.T) {
 			select {
 			case <-timeout.Done():
 				assert.FailNow(t, "timed out")
-			case event := <-sub.Events():
-				assert.IsType(t, events.Event{}, event)
-				actualHeight := event.(events.Event).Height
-				if actualHeight == 0 {
-					assert.Equal(t, expectedEventCount, eventCount)
-					return
-				}
-				assert.Equal(t, newBlock.Height, actualHeight)
-				eventCount++
-			}
-		}
-	}).Repeat(repeats))
-
-	t.Run("match tm.event=Tx", testutils.Func(func(t *testing.T) {
-		setup()
-
-		bus.FetchEvents(context.Background())
-		query.MatchesFunc = func(events map[string][]string) (bool, error) {
-			types := events[tm.EventTypeKey]
-
-			for _, t := range types {
-				if t == tm.EventTx {
-					return true, nil
-				}
-			}
-			return false, nil
-		}
-		sub, err := bus.Subscribe(query)
-		assert.NoError(t, err)
-
-		newBlock := &coretypes.ResultBlockResults{
-			Height:           rand.PosI64(),
-			BeginBlockEvents: randomEvents(rand.I64Between(0, 10)),
-			TxsResults:       randomTxResults(rand.I64Between(1, 10)),
-			EndBlockEvents:   randomEvents(rand.I64Between(0, 10)),
-		}
-
-		endMarkerBlock := &coretypes.ResultBlockResults{
-			Height:           0,
-			BeginBlockEvents: randomEvents(rand.I64Between(3, 10)),
-			TxsResults:       randomTxResults(rand.I64Between(1, 10)),
-			EndBlockEvents:   randomEvents(rand.I64Between(3, 10)),
-		}
-		newBlocks <- newBlock
-		newBlocks <- endMarkerBlock
-
-		timeout, cancel := context.WithTimeout(context.Background(), 1*time.Second)
-		defer cancel()
-
-		expectedEventCount := 0
-		for _, result := range newBlock.TxsResults {
-			expectedEventCount += len(result.Events)
-		}
-		var eventCount int
-		for {
-			select {
-			case <-timeout.Done():
-				assert.FailNow(t, "timed out")
-			case event := <-sub.Events():
-				assert.IsType(t, events.Event{}, event)
-				actualHeight := event.(events.Event).Height
```

Comment:
Why Can we remove these test cases?

Reply:
because the bus no longer supports tm queries

Reply:
it's using a simple predicate function instead now